util/i2csvg/README.md

```diff
@@ -0,0 +1,159 @@
+|'tag'| In single quotes: Tag for the value that is drawn in output |
+
+Lines of the input may consist of a single flag+value per line or a comma separated list of flag+value pairs.
+
+### Hexdump source format
+
+In the hexdump source format tool processes the raw 13-bit data that is written to the I2C fifo.
+This is normally in hex (preceded with 0x) but the tool will accept binary, octal or decimal.
+Lines of input may consist of a single hex value, multiple values separated by commas (and optionally whitespace) or multiple values separated by whitespace.
+
+### Output file format
+
+The output file generated depends on the options specified.
+
+|Type | Options|
+|-----|--------|
+|.svg  | If multiple output files are generated or debug/text not selected |
+|.html | If debug/text is selected and svg is generated |
+|.txt  | If debug/test is selected and no svg is generated (--nosvg) |
+
+The example commands assume `$REPO_TOP` is set to the toplevel directory
+of the repo.
+
+### Setup
+
+The i2ctool does not depend on any other packages.
+
+### Examples using i2csvg.py
+
+The i2csvg/examples directory contains some example I2C files.
+For simple examples text can be piped in to the tool using echo.
+
+A simple I2C write consists of a Start with Address and direction bit of zero, and a single data byte (4 in the example) with a Stop.
+
+The baseline tool will generate an svg file (which can be viewed in a browser):
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py > out.svg
+```
+
+The text version (which is just the input with more structured formatting) can be viewed in the terminal by suppressing the svg output:
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py --text --nosvg
+S.... A0
+.P... 0x4
+```
+
+The debug version is intended for tool development:
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py --debug --nosvg
+I2cOp(read=False, rcont=False, start=True, stop=False, nackok=False, mvalue=False, adr=True, size=10, fbyte=0, tag=None)
+I2cOp(read=False, rcont=False, start=False, stop=True, nackok=False, mvalue=False, adr=False, size=10, fbyte=4, tag=None)
+```
+
+If svg generation is not suppressed then the text or debug output and the svg are generated in an HTML file.
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py --text > out.html
+```
+
+The smbus directory contains a files with each of the commands listed in SMBus 3.0 specification.
+These can be used to generate an HTML file with all the transactions expanded.
+
+```console
+$ cd $REPO_TOP/util
+$ ./i2csvg.py i2csvg/smbus/*.txt > out.html
+```
+
+Alternatively, these can all be converted to individual svg files (for example for inclusion in documents) with the --multiout command.
+The output filename matches the input filename with the extension replaced with .svg
+Note that if the --text or --debug flags are given then the output and extension will be html
```

Comment:
probably want all command line arguments in `backquote` like `--text` and `--debug`